pkg/config/config.go

```diff
@@ -1775,14 +1774,17 @@ func ResolveSecrets(config Config, secretResolver secrets.Component, origin stri
 			return fmt.Errorf("unable to marshal configuration to YAML to decrypt secrets: %v", err)
 		}
 
-		finalYamlConf, err := secretResolver.Decrypt(yamlConf, origin)
+		err = secretResolver.DecryptWithCallback(
+			yamlConf,
+			origin,
+			func(yamlPath []string, value any) {
+				settingName := strings.Join(yamlPath, ".")
+				log.Debugf("replacing handle '%s' for setting '%s' with secret value", config.Get(settingName), settingName)
```

Comment:
Maybe this?
```suggestion
				log.Debugf("Resolving configuration field '%s' secret value ('%s')", settingName, config.Get(settingName))
```

I'm a bit curious if we consider the secret path sensitive for logs.

Reply:
Also, are we sure that config will not contain a resolved secret value when doing a `config.Get`?